include/linux/tee_drv.h

```diff
@@ -42,11 +43,15 @@ struct tee_shm_pool;
  * @teedev:	pointer to this drivers struct tee_device
  * @list_shm:	List of shared memory object owned by this context
  * @data:	driver specific context data, managed by the driver
+ * @registered: reference counter for this structure
+ * @releasing:  flag that indicates if context is being released right now
  */
 struct tee_context {
 	struct tee_device *teedev;
 	struct list_head list_shm;
 	void *data;
+	struct kref refcount;
+	bool releasing;
```

Comment:
@jenswi-linaro, I described that in this and next comment: https://github.com/linaro-swg/linux/pull/29#issuecomment-317473581